tools/etcd-tls-setup/bin/run.sh

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env sh
+set -e
+
+CURRENT_DIR="$(CDPATH= cd -- "$(dirname -- "$0")" && pwd)"
+
+NAMESPACE=${NAMESPACE:-"kyma-system"}
+
+# ETCD_CLUSTER_NAME - should be set as value of EtcdCluster CR's name
+
+caSecretName="${ETCD_CLUSTER_NAME}-etcd-ca-tls"
+serverSecretName="${ETCD_CLUSTER_NAME}-etcd-server-tls"
+peerSecretName="${ETCD_CLUSTER_NAME}-etcd-peer-tls"
+clientSecretName="${ETCD_CLUSTER_NAME}-etcd-client-tls"
+
+tempPath="/tmp"
+configPath="${CURRENT_DIR}/../config"
+
+sed "s#__ETCD_CLUSTER_NAME__#$ETCD_CLUSTER_NAME#g
+    s#__NAMESPACE__#$NAMESPACE#g
+    " ${configPath}/peer.json.tpl > ${configPath}/peer.json
+sed "s#__ETCD_CLUSTER_NAME__#$ETCD_CLUSTER_NAME#g
+    s#__NAMESPACE__#$NAMESPACE#g
+    " ${configPath}/server.json.tpl > ${configPath}/server.json
+
+if [[ "$(kubectl get secret ${caSecretName} -n ${NAMESPACE})" ]]; then
+    echo "CA certificate found. Downloading..."
+    kubectl get secret ${caSecretName} -n ${NAMESPACE} -o=jsonpath='{.data.ca\.pem}' | base64 -d > ${tempPath}/ca.pem
+    kubectl get secret ${caSecretName} -n ${NAMESPACE} -o=jsonpath='{.data.ca-key\.pem}' | base64 -d > ${tempPath}/ca-key.pem
+else
+    echo "CA certificate does not exist. Generating new one..."
+    cfssl gencert -initca ${configPath}/ca-csr.json | cfssljson -bare ${tempPath}/ca -
+    kubectl create secret generic ${caSecretName} --from-file=${tempPath}/ca.pem --from-file=${tempPath}/ca-key.pem --dry-run -o yaml > ${tempPath}/ca-secret.yaml
+    kubectl apply -f ${tempPath}/ca-secret.yaml
+fi
+
+
+echo "generating etcd peer certs ==="
+cfssl gencert -ca=${tempPath}/ca.pem -ca-key=${tempPath}/ca-key.pem -config=${configPath}/ca-config.json -profile=peer ${configPath}/peer.json | cfssljson -bare ${tempPath}/peer
```

Comment:
suggestion: maybe it can be just a function with param `peer`,`server`, `client`  thanks to that you will make it DRY